File: app/src/main/java/ar/com/wolox/android/example/ui/home/HomeFragment.java
package ar.com.wolox.android.example.ui.home;

import ar.com.wolox.android.R;
import ar.com.wolox.android.example.model.User;
import ar.com.wolox.android.example.ui.example.ExamplePresenter;
import ar.com.wolox.android.example.ui.login.ILoginView;
import ar.com.wolox.wolmo.core.fragment.WolmoFragment;

/**
 *
 */

public class HomeFragment extends WolmoFragment<ExamplePresenter> implements ILoginView {

    @Override
    public int layout() {
        return R.layout.fragment_home;
    }

    @Override
    public void init() {
    }

    @Override
    public void onUsernameSaved() {
    }

    @Override
    public void onUsernameValidated(User user) {

    }
}

File: app/src/main/java/ar/com/wolox/android/example/ui/home/HomeFragment.kt
package ar.com.wolox.android.example.ui.home

import android.graphics.Color
import android.support.v4.app.Fragment
import android.support.v4.util.Pair
import ar.com.wolox.android.R
import ar.com.wolox.android.example.ui.home.news.NewsFragment
import ar.com.wolox.android.example.ui.home.profile.ProfileFragment
import ar.com.wolox.wolmo.core.adapter.viewpager.SimpleFragmentPagerAdapter
import ar.com.wolox.wolmo.core.fragment.WolmoFragment
import ar.com.wolox.wolmo.core.presenter.BasePresenter
import kotlinx.android.synthetic.main.fragment_home.*
import kotlinx.android.synthetic.main.fragment_viewpager.vViewPager
import javax.inject.Inject

class HomeFragment @Inject constructor() : WolmoFragment<BasePresenter<Any>>() {

    private lateinit var fragmentPagerAdapter: SimpleFragmentPagerAdapter

    override fun layout(): Int = R.layout.fragment_home

    override fun init() {

        fragmentPagerAdapter = SimpleFragmentPagerAdapter(childFragmentManager)
        fragmentPagerAdapter.addFragments(
                Pair<Fragment, String>(NewsFragment.newInstance(), getString(R.string.news)),
                Pair<Fragment, String>(ProfileFragment.newInstance(), getString(R.string.profile)))
        vViewPager.adapter = fragmentPagerAdapter

        vTabs.setupWithViewPager(vViewPager)

        vTabs.setTabTextColors(Color.parseColor("#7E7E7E"), Color.parseColor("#8DC63F"))
        vTabs.getTabAt(0)?.setIcon(R.drawable.ic_news_on_off)
        vTabs.getTabAt(1)?.setIcon(R.drawable.ic_profile_on_off)
    }
}